algebird-core/src/main/scala/com/twitter/algebird/CorrelationGroup.scala

```diff
@@ -0,0 +1,158 @@
+package com.twitter.algebird
+
+object Correlation {
+  def apply(x: (Double, Double), weight: Double): Correlation =
+    Correlation(c2 = 0, m2x = 0, m2y = 0, m1x = x._1, m1y = x._2, weight)
+
+  def apply(x: (Double, Double)): Correlation =
+    apply(x, 1.0)
+
+  implicit val group: Group[Correlation] = CorrelationGroup
+
+  /**
+   * When combining averages, if the counts sizes are too close we
+   * should use a different algorithm.  This constant defines how
+   * close the ratio of the smaller to the total count can be:
+   */
+  private val STABILITY_CONSTANT = 0.1
+
+  /**
+   * Given two streams of doubles (weightN, an) and (weightK, ak) of form (weighted count,
+   * mean), calculates the mean of the combined stream.
+   *
+   * Uses a more stable online algorithm which should be suitable for
+   * large numbers of records similar to:
+   * http://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm
+   *
+   * This differs from the implementation in MomentsGroup.scala only in that here, the counts are weighted, and are
+   * thus doubles instead of longs
+   */
+  def getCombinedMean(weightN: Double, an: Double, weightK: Double, ak: Double): Double =
+    if (weightN < weightK) getCombinedMean(weightK, ak, weightN, an)
+    else
+      (weightN + weightK) match {
+        case 0L                              => 0.0
```

Comment:
shouldn't this be `0.0` not `0L`?

Reply:
Good catch. This was lazy copy-pasting on my part.